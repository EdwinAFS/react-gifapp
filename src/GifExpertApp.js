import React, { Fragment, useState } from "react";
import { CategoryAdd } from "./components/CategoryAdd";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {
	
	const [categories, setCategories] = useState(defaultCategories);

	return (
		<Fragment>
			<h2> GifExperApp</h2>
			<hr />
			<CategoryAdd setCategories={setCategories}/>
			<ol>
				{
					categories.map( category => (
						<GifGrid 
							key = { category }
							category = { category }
						/>
					))
				}
			</ol>
		</Fragment>
	);
};
