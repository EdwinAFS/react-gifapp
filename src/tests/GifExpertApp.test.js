import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Test para el componente GifGrid', () => {

	test('Deberia mostrar la interfaz GifExperApp', ()=>{
		 const wrapper = shallow(<GifExpertApp />);
		 expect(wrapper).toMatchSnapshot();
	});

	test('Deberia mostrar una lista de categorias', ()=>{

		const categories = ['Vikings'];
		const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

		expect( wrapper.find('GifGrid').length ).toBe( categories.length ); 
		
   });

});