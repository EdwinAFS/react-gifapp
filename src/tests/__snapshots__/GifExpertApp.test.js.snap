// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para el componente GifGrid Deberia mostrar la interfaz GifExperApp 1`] = `
<Fragment>
  <h2>
     GifExperApp
  </h2>
  <hr />
  <CategoryAdd
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
